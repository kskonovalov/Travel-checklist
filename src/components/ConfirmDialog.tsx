import React from 'react';
import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions,
  Button,
  TextField
} from '@material-ui/core';
import { useHistory } from 'react-router-dom';

interface IProps {
  open: boolean;
  onClose: () => void;
}

const ConfirmDialog = ({ onClose, open }: IProps) => {
  const history = useHistory();

  return (
    <Dialog onClose={onClose} open={open}>
      <DialogTitle>Вы уверены?</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          Это действие создаст новый список дел, старый список останется
          доступен по ссылке:
        </DialogContentText>
        <TextField fullWidth={true} value={`${window.location.href}`} />
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} color="primary">
          Отменить
        </Button>
        <Button
          onClick={() => {
            localStorage.removeItem('listID');
            localStorage.removeItem('tasks');
            history.push(`/`);
          }}
          color="primary"
          variant="outlined"
        >
          Создать новый
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default ConfirmDialog;
