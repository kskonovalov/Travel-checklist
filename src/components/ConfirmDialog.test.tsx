import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

import ConfirmDialog from './ConfirmDialog';

describe('Component: ConfirmDialog', () => {
  test('should render component description', () => {
    // @ts-ignore
    const component = shallow(
      <ConfirmDialog open={true} onClose={() => {}} onConfirm={() => {}} />
    );
    const description = component.find('#alert-dialog-description');
    expect(description.length).toBe(1);
  });
  test('should render component buttons', () => {
    // @ts-ignore
    const component = shallow(
      <ConfirmDialog open={true} onClose={() => {}} onConfirm={() => {}} />
    );
    const buttons = component.find('WithStyles(ForwardRef(Button))');
    expect(buttons.length).toBe(2); // confirm & cancel
    // console.log(component.debug());
  });
});
