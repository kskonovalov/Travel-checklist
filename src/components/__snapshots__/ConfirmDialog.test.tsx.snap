// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ConfirmDialog Should match snapshot 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    Вы уверены?
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))
      id="alert-dialog-description"
    >
      Это действие создаст новый список дел, старый список останется доступен по ссылке:
    </WithStyles(ForwardRef(DialogContentText))>
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      value="http://localhost/"
    />
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
    >
      Отменить
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
      variant="outlined"
    >
      Создать новый
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
